py-polars/polars/eager/series.py

```diff
@@ -2089,6 +2089,120 @@ def mode(self) -> "Series":
         """
         return wrap_s(self._s.mode())
 
+    def sin(self) -> "Series":
+        """
+        Compute the element-wise value for Trigonometric sine.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> s = pl.Series("a", np.array((0., np.pi/2., np.pi)))
+        >>> s.sin()
+        shape: (3,)
+        Series: 'a' [f64]
+        [
+                0.0
+                1
+                1.2246467991473532e-16
+        ]
+        """
+        return wrap_s(numpy_to_pyseries(self.name, np.sin(self.to_numpy())))
+
+    def cos(self) -> "Series":
+        """
+        Compute the element-wise value for Trigonometric cosine.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> s = pl.Series("a", np.array((0., np.pi/2., np.pi)))
+        >>> s.cos()
+        shape: (3,)
+        Series: 'a' [f64]
+        [
+                1
+                6.123233995736766e-17
+                -1e0
+        ]
+        """
+        return wrap_s(numpy_to_pyseries(self.name, np.cos(self.to_numpy())))
```

Comment:
This has an unneeded allocation. So we first go to numpy, which may allocate (only if null values). And then we need to copy the values from numpy to polars, which is again an allocation.

We should just use `np.cos(self)` and then add `# type: ignore` because `mypy` doesn't understand we get a `Series` instead of a numpy array as result.

Polars implements numpy universal functions and makes sure it doesn't do unnecessary allocations.

Reply:
Thanks for taking the time to explain and advise. 

I think that's what I initially did but ultimately led to a path of trying to satisfy mypy to not add the ignore comments. 

Will make the adjustment and then we should be all good.